### ISSUE TITLE
Manual preset runtime

### ISSUE THREAD

**Issue by ingo916** [sebr/bhyve-home-assistant#187]
**Describe the bug**
 bhyve: set_manual_preset_runtime will not update the preset runtime. 
**Expected behaviour**
running the service 
service: bhyve.set_manual_preset_runtime
data:
  entity_id: switch.drip_zone
  minutes: 30
SHOULD CHANGE THE PRESET TO 30 MINUTES

**BHyve devices**
Orbit 57950 B-hyve Smart 12-Zone Indoor/Outdoor Sprinkler Controller

**Device Diagnostics**

![image](https://github.com/sebr/bhyve-home-assistant/assets/41309095/ee23ca67-05e9-4a95-bb07-f9be34269196)
![image](https://github.com/sebr/bhyve-home-assistant/assets/41309095/23f4f3a6-cce2-4b45-b64d-0272ceaa5ee9)

**Comment by bsp9493** [comment]
I am also having this issue.

I have been using this integration for years and this was never a problem…

activating the entity via another integration (irrigation controller) the device would run for the time defined in that integration)

nit sure if I missed something in a recent update, but now, when the switch entity is called (to run for 16 mins for example), I t automatically shuts off after the default runtime of the orbit device (which is 10 mins)


I have not been able to update the default runtime and have it stick.

I have the newest firmware in my orbit controller and there is also NO option to change it there:(


Love your integration, t has worked flawlessly for year, problem just started this season.

your help is appreciated

**Comment by kkellner** [comment]
Same problem here.  First time using `set_manual_preset_runtime` but it doesn't seem to be doing anything.  No info in logs that that call failed.  I get a green checkbox when I run in `Developer Tools --> Services` but the attribute never gets changed.

**Comment by sebr** [comment]
Works fine for me with a Hose Timer (Model HT25).

@ingo916 I suspect this is because your device (Orbit 57950) has multiple zones? I don't really know how to unblock this as I don't have the hardware to test / investigate.

> nit sure if I missed something in a recent update, but now, when the switch entity is called (to run for 16 mins for example), I t automatically shuts off after the default runtime of the orbit device (which is 10 mins)

@bsp9493 - this sounds completely separate, and not related to the `set_manual_preset_runtime` service call. Please open a new issue if so.

**Comment by kkellner** [comment]
I have the 16-station XR (Model WT26-0001) which is not working.

![image](https://github.com/sebr/bhyve-home-assistant/assets/9540241/14027c04-14c5-4081-827d-f1c79667f86d)

@sebr any debugging logging I can turn on and provide let me know.

**Comment by sebr** [comment]
Can you tell me...

In the official app, if you navigate to: `My Bhyve` > `Devices` > Your Device > `Preset Runtime` what do you see? Is it one option for the entire device or are you able to choose a preset runtime per zone?

The service call currently sets the preset runtime for the device (it doesn't send a zone in the payload) so this is my best guess at the moment.

**Comment by kkellner** [comment]
Looks like that option does not show on XR models.

![image](https://github.com/sebr/bhyve-home-assistant/assets/9540241/fcc23fac-75ac-4371-8ec9-483a1f6532d8 )

**Comment by bsp9493** [comment]
I can confirm this option is also NOT available on6 station indoor/outdoor timer8 station indoor timerBoth have a 10min defaultdOn May 31, 2023, at 7:25 AM, Kurt Kellner ***@***.***> wrote:﻿
Looks like that option does not show on XR models.

**Comment by mesencephalon** [comment]
I have Orbit 24639 B-hyve XD 4 port smart hose watering timer and  manual preset runtime option doesn't exist in bhyve app settings. I tried to use set_manual_preset_runtime call in home assistant automation but it never works. I can only set manual watering time in bhyve app for manual watering on home screen of bhyve app. Any solution to set manual runtime on home assistant with this model of bhyve?

**Comment by krusej23** [comment]
Wondering if this was ever figured out or if Orbit pulled it from the API.

**Comment by Shredder5262** [comment]
I have the Orbit 24639 B-hyve XD 4-Port, i tried using the call service. It does nothing for me. The manual preset time option in the app also does not exist for me.

**Comment by Shredder5262** [comment]
Any feedback on this would be appreciated, this is really the only thing that i want this device for is to be able to turn it on from within home assistant. Yes I'm aware i can use the app for that, but it defeats the purpose of needing home assistant for that. Is this int. still being supported?

**Comment by sebr** [comment]
@Shredder5262 as described earlier in this thread, some devices do not support the "manual preset run time". I am going to update the readme with this information so that it is clearer.

The workaround (or perhaps, the proper solution) is to use the [`bhyve.start_watering` service call](https://github.com/sebr/bhyve-home-assistant?tab=readme-ov-file#services) with the required number of minutes to run.

**Comment by krusej23** [comment]
@Shredder5262 May this will help. I use a button on my dashboard and use the call service option.
<img width="398" alt="Screenshot 2024-05-27 at 6 15 50 PM" src="https://github.com/sebr/bhyve-home-assistant/assets/10549869/c500f193-c5a2-4ce4-9bb8-34b232f33cd5">

**Comment by Shredder5262** [comment]
> @Shredder5262 May this will help. I use a button on my dashboard and use the call service option. <img alt="Screenshot 2024-05-27 at 6 15 50 PM" width="398" src="https://private-user-images.githubusercontent.com/10549869/334225402-c500f193-c5a2-4ce4-9bb8-34b232f33cd5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc4NjUwOTgsIm5iZiI6MTcxNzg2NDc5OCwicGF0aCI6Ii8xMDU0OTg2OS8zMzQyMjU0MDItYzUwMGYxOTMtYzVhMi00Y2U0LTliYjgtMzRiMjMyZjMzY2Q1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA4VDE2Mzk1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkYmE3YzAyYzdlYzU3MmQ3YjFiNDVmMzRjZjk5MmVhYTAwOWRlNGQyOTUyM2RmNWIzMjNkZDFhZmE3MGViN2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hMy3qLBTf9Y250BFP6U4424ucrPFbk-iS_a8Nd9JdHQ">

ok, that seems to have done something, I'll try to work with that, thank you for the suggestion.
